test/e2e_env/kubernetes/healthcheck/virtual_probes.go

```diff
@@ -37,26 +32,18 @@ func VirtualProbes() {
 		Expect(err).To(Succeed())
 	})
 
-	PollPodsReady := func() error {
-		pods, err := k8s.ListPodsE(env.Cluster.GetTesting(), env.Cluster.GetKubectlOptions(namespace),
-			metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", name)})
-		if err != nil {
-			return err
-		}
-		for _, p := range pods {
-			err := k8s.WaitUntilPodAvailableE(env.Cluster.GetTesting(), env.Cluster.GetKubectlOptions(namespace), p.GetName(), 0, 0)
-			if err != nil {
-				return err
-			}
-		}
-		return nil
-	}
-
 	It("should deploy test-server with probes", func() {
 		// Sample pod readiness to ensure they stay ready to at least 10sec.
-		for i := 0; i < 10; i++ {
-			time.Sleep(time.Second)
-			Expect(PollPodsReady()).To(Succeed())
-		}
+		Eventually(func(g Gomega) {
+			pods, err := PodsAvailable(env.Cluster, name, namespace)
```

Comment:
Is this not `WaitNumPods`?

Reply:
not exactly, but got rid of PodsAvailable anyways. Thanks for pointing this out